scripts/test_requirements.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+#
+# Check whether requirements*.in been changed without updating requirements*.txt
+#
+# We want to make sure that we are running `pip-compile` whenever we change our
+# Python requirements.
+#
+# In this repo the requirements files are generated from `*.in` files. This
+# script checks whether an in file has changed but the corresponding requirements
+# file has not.
+#
+# Returns 1 if `pip-compile` needs to be run, otherwise 0
+
+
+# Requirements files to check
+REQUIREMENTS_FILES=${@:-requirements.txt requirements-dev.txt}
+
+# Git reference to compare against
+BASE_COMMIT=${BASE_COMMIT:-master}
+HEAD_COMMIT=${HEAD_COMMIT:-HEAD}
+
+# Return 1 if file(s) has changed relative to BASE_COMMIT, otherwise 0
+test_unchanged() {
+  git diff --quiet $BASE_COMMIT $HEAD_COMMIT -- $@
```

Comment:
```suggestion
  git diff --quiet $BASE_COMMIT $HEAD_COMMIT -- "$@"
```

Tiny suggestion: guard against spaces in filename. But you'd need to change the rest of this script too for any mention of the file so happy if you disregard